test/task-affinity.jl

```diff
@@ -0,0 +1,577 @@
+                scope_intersect          = compute_scope_intersect
+                scope_rand               = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+                result_scope_intersect   = Dagger.UnionScope(scope_b..., scope_c...)
+
+                task1 = Dagger.@spawn compute_scope=compute_scope_intersect result_scope=result_scope_intersect f(10);                  wait(task1); fetch(task1)
+                task2 = Dagger.@spawn scope=scope_intersect result_scope=result_scope_intersect f(20);                                  wait(task2); fetch(task2)
+                task3 = Dagger.@spawn compute_scope=compute_scope_intersect scope=scope_rand result_scope=result_scope_intersect f(30); wait(task3); fetch(task3)
+
+                @test get_compute_scope(task1) == get_compute_scope(task2) == get_compute_scope(task3) == compute_scope_intersect
+                @test get_result_scope(task1)  == get_result_scope(task2)  == get_result_scope(task3)  == result_scope_intersect
+
+                execution_scope1   = get_execution_scope(task1)
+                execution_scope2   = get_execution_scope(task2)
+                execution_scope3   = get_execution_scope(task3)
+                @test execution_scope1 in intersect_scopes(execution_scope1, compute_scope_intersect, result_scope_intersect) &&
+                    execution_scope2 in intersect_scopes(execution_scope2, compute_scope_intersect, result_scope_intersect) &&
+                    execution_scope3 in intersect_scopes(execution_scope3, compute_scope_intersect, result_scope_intersect) 
+            end
+        end
+
+        @testset "compute_scope and result_scope Tests without Intersection" begin
+            if length(availscopes) >= 2
+                n = cld(numscopes, 2)
+
+                scope_a = availscopes[1:n]
+                scope_b = availscopes[n+1:end]
+
+                compute_scope_no_intersect = Dagger.UnionScope(scope_a...)
+                scope_no_intersect         = Dagger.UnionScope(scope_a...)
+                scope_rand                 = Dagger.UnionScope(rand(availscopes, rand(1:length(availscopes))))
+                result_scope_no_intersect  = Dagger.UnionScope(scope_b...)
+
+                task1 = Dagger.@spawn compute_scope=compute_scope_no_intersect result_scope=result_scope_no_intersect f(10);                  wait(task1)
+                task2 = Dagger.@spawn scope=scope_no_intersect result_scope=result_scope_no_intersect f(20);                                  wait(task2)
+                task3 = Dagger.@spawn compute_scope=compute_scope_no_intersect scope=scope_rand result_scope=result_scope_no_intersect f(30); wait(task3)
+
+                @test get_compute_scope(task1) == get_compute_scope(task2) == get_compute_scope(task3) == compute_scope_no_intersect
+                @test get_result_scope(task1)  == get_result_scope(task2)  == get_result_scope(task3)  == Dagger.InvalidScope
```

Comment:
```suggestion
                @test get_result_scope(task1)  == get_result_scope(task2)  == get_result_scope(task3)  && get_result_scope(task1) isa Dagger.InvalidScope
```

I think this is what you really mean? That the resulting intersection is empty?